packages/datatrak-web/src/api/mutations/useLogin.ts

```diff
@@ -12,12 +12,19 @@ type LoginCredentials = {
   email: string;
   password: string;
 };
+
+function hasFrom(state: unknown): state is { from: string } {
+  if (state !== null && typeof state === 'object' && 'from' in state) {
+    return typeof state.from === 'string';
+  }
+  return false;
+}
 export const useLogin = () => {
   const queryClient = useQueryClient();
   const navigate = useNavigate();
   const location = useLocation();
 
-  const { from } = location.state as { from: string };
```

Comment:
I was getting an error on first login where state was null so I added some more robust checking.

Reply:
We could probably simplify this to be just `location.state && location.state.from`

Reply:
I tried that but I get a TS error. I could cast it using `as` but Rohan said that it's not good practise to cast and that I should use typeguards so I'm trying that out. It seems a bit verbose so maybe there is a more concise way of doing it.